scrapy/link.py

```diff
@@ -28,6 +28,7 @@ def __init__(self, url, text='', fragment='', nofollow=False):
         self.text = text
         self.fragment = fragment
         self.nofollow = nofollow
+        self.encoding = encoding
```

Comment:
In Python 2.x all URLs were bytes, so we required link.url to be bytes. For backwards compatibility, and because of missing unicode support in url-related functions in Python 2.x, link.url is still required to be bytes in Python 2.x